deeppavlov/core/agent_v2/agent.py

```diff
@@ -22,14 +23,11 @@ def __call__(self, utterances: Sequence[str], user_telegram_ids: Sequence[Hashab
         should_reset = [utterance == TG_START_UTT for utterance in utterances]
         # here and further me stands for mongoengine
         me_users = self.state_manager.get_users(user_telegram_ids, user_device_types)
-        me_utterances = self.state_manager.get_utterances(utterances, me_users, date_times)
-        me_dialogs = self.state_manager.get_dialogs(me_users, me_utterances, locations, channel_types, should_reset)
+        me_dialogs = self.state_manager.get_dialogs(me_users, locations, channel_types, should_reset)
+        self.state_manager.add_user_utterances(me_dialogs, utterances, date_times)
```

Comment:
Let's make void methods from `StateManager.add_user_utterances` and `StateManager.add_bot_utterances`. Utterances can be easily obtained than from the modified dialog object (`dialog.utterances`).